api/types/types.go

```diff
@@ -149,9 +149,24 @@ type Version struct {
 	GoVersion     string
 	Os            string
 	Arch          string
-	KernelVersion string `json:",omitempty"`
-	Experimental  bool   `json:",omitempty"`
-	BuildTime     string `json:",omitempty"`
+	KernelVersion string         `json:",omitempty"`
+	Experimental  bool           `json:",omitempty"`
+	BuildTime     string         `json:",omitempty"`
+	DockerVersion *DockerVersion `json:",omitempty"`
+}
+
+// DockerVersion contains docker version info and
+// is returned by:
+// GET "/version"
+type DockerVersion struct {
+	Type    string // CE, EE...
+	Version string // 17.01, 18.04...
+	Flavor  string // mac, win...
+	Patch   string // 0, 1...
+}
+
+func (ev *DockerVersion) String() string {
+	return strings.Join([]string{ev.Type, ev.Version, ev.Flavor, ev.Patch}, "-")
```

Comment:
Ok I'm really nitpicking here, but can we have `CE 17.01-mac0` instead? So:

    return fmt.Sprintf("%s %s-%s%s", ev.Type, ev.Version, ev.Flavor, ev.Patch)

:innocent: